astroid/modutils.py

```diff
@@ -138,21 +139,19 @@ def _handle_blacklist(blacklist, dirnames, filenames):
             filenames.remove(norecurs)
 
 
-_NORM_PATH_CACHE: Dict[str, str] = {}
+@lru_cache()
+def _cache_normalize_path_(path: str) -> str:
+    return _normalize_path(path)
 
 
 def _cache_normalize_path(path: str) -> str:
```

Comment:
Can't we cache this functions?
What does `_normalize_path(path)` do when `path` is `""`?

Reply:
It returns current working directory. (That doesn't seem appropriate to cache, right?)

Reply:
It's `os.path.normcase(os.path.realpath(path))` right? Would that ever change during a `astroid` or `pylint` run?

Reply:
I think we need to imagine long-running processes and IDEs. I'd say let's not introduce a behavior change on a maintenance PR.